File: components/amp/example-js/ESP32/examples/human_detector/ir.js
class _IR {
    constructor(gpioObj) {
        this.gpioInstance = gpioObj;
    };

    irDetect() {
	var value = this.gpioInstance.read();
	return value;
    };
}

export function IR(gpioObj) {
    return new _IR(gpioObj);
}

File: haas_lib_bundles/python/docs/examples/human_detector/haaseduk1/code/ir.py
from driver import GPIO

class IR(object):

    def __init__(self, gpioObj):
        self.gpioObj = None
        if not isinstance(gpioObj, GPIO):
            raise ValueError("parameter is not a GPIO object")

        self.gpioObj = gpioObj

    def irDetect(self):
        if self.gpioObj is None:
            raise ValueError("invalid GPIO object")

        value = self.gpioObj.read()
        return value
